services/src/main/java/org/keycloak/authorization/authorization/AuthorizationTokenService.java

```diff
@@ -802,5 +806,100 @@ RealmModel getRealm() {
         ClientConnection getClientConnection() {
             return clientConnection;
         }
+
+        public void addPermissions(List<String> permissionList, String permissionResourceFormat) {
+            if (permissionResourceFormat == null) {
+                permissionResourceFormat = "id";
+            }
+
+            switch (permissionResourceFormat) {
+                case "id":
+                    addPermissionsById(permissionList);
+                    break;
+                case "uri":
+                    addPermissionsByUri(permissionList);
+                    break;
+                default:
```

Comment:
L822-L824 are needed?

Reply:
Not required. If parameter values other than the above two are specified, it's a bad request, so I was thinking of leaving room to incorporate some error handling in the future.
I delete them.